### PR TITLE
Fix bug with trinity not shutting down cleanly

### LINKED ISSUE #872: Trinity doesn't shut down cleanly
* Trinity version: 6493c94656b907c
* OS: osx/linux

### What is wrong?

Ctrl-C no longer cleanly exits. You need a second Ctrl-C to get out. Occasionally, it leaves dirty IPC files, or even hanging processes that never exit.

### How can it be fixed

:man_shrugging: 

I ran `git bisect` and narrowed it down to the above commit. cc @cburgdorf  @pipermerriam

### PR DESCRIPTION
**Pull request by pipermerriam**
fixes #872 

### What was wrong?

When the `PluginManager` was moved into a service I failed to tie the shutdown process of the plugin manager to the shutdown process of the service and that caused Trinity to hang on shutdown since it didn't propagate the shutdown request to the various plugins.

### How was it fixed?

Made sure to call `PluginManager.shutdown_blocking` in the `BaseService._cleanup`.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/trinity/blob/master/newsfragments/README.md)

[//]: # (See: https://trinity-client.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/trinity/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: p2p/peer_pool.py [hunk 1]
```diff
@@ -201,10 +201,13 @@ def setup_peer_backends(self) -> Tuple[BasePeerBackend, ...]:
...
-            await self.wait(asyncio.gather(*(
-                self._add_peers_from_backend(backend)
-                for backend in self.peer_backends
-            )))
+            try:
+                await self.wait(asyncio.gather(*(
+                    self._add_peers_from_backend(backend)
+                    for backend in self.peer_backends
+                )))
+            except OperationCancelled:
```
Comment (pipermerriam) [inline p2p/peer_pool.py]: This catches a shutdown exception I got during experimentation with this.

File: trinity/_utils/mp.py [hunk 2]
```diff
@@ -32,12 +32,14 @@ def async_method(method_name: str) -> Callable[..., Any]:
             method_name,
             args,
         )
+    method.__name__ = method_name
```
Comment (pipermerriam) [inline trinity/_utils/mp.py]: This makes these methods have an identifiable name when they end up in terminal output.

File: trinity/sync/common/headers.py [hunk 3]
```diff
@@ -165,10 +165,10 @@ def __init__(self,
             # validate that parents and children match
             pairs = tuple(zip(parents, children))
             try:
-                validate_pair_coros = [
+                validate_pair_coros = (
```
Comment (pipermerriam) [inline trinity/sync/common/headers.py]: lazy-ify this so that the none of the functions actually get called until they are consumed in `asyncio.gather`

Comment (cburgdorf) [reply trinity/sync/common/headers.py]: Might be worth putting this into a separate commit for improved clarity

File: trinity/event_bus.py [hunk 4]
```diff
@@ -54,17 +54,21 @@ def __init__(self,
             # start the background process that tracks and propagates available
             # endpoints to the other connected endpoints
             self.run_daemon_task(self._track_and_propagate_available_endpoints())
+            self.run_daemon_task(self._handle_shutdown_request())
```
Comment (cburgdorf) [inline trinity/event_bus.py]: Sorry for not spotting this in my initial review.

File: trinity/sync/common/headers.py (hunk 3 above)
Comment (carver) [reply trinity/sync/common/headers.py]: AFAICT, this doesn't have an impact one way or another. It expands the whole collection as an early step of invoking `gather()`. No need to change it back, I just want to make sure we all agree on what's happening here. :)

Comment (pipermerriam) [reply trinity/sync/common/headers.py]: :+1: 

This was partially me stabbing at a warning I was getting about coroutines not being awaited and I thought *maybe* it was one of these getting called but not getting awaited due to a shutdown interrupting. (I don't think it actually fixed it)

**Pull request merged**